medvis/dicom/src/errorlogging.cpp

```diff
@@ -0,0 +1,240 @@
+                }
+                buffer.clear();
+            } else if (!Traits::eq_int_type(ch, Traits::eof())) {
+                buffer.push_back(static_cast<CharT>(ch));
+            }
+        } catch (std::exception &ex) {
+            // error while logging, print what we've got so far
+            inviwo::LogCentral::getPtr()->log("Gdcm Volume Importer", loglevel, LogAudience::User,
+                                              "<GDCM library>", "", 0, buffer);
+            buffer.clear();
+            // and tell the user what hargspened
+            LogError("Exception while logging Gdcm: " << ex.what());
+            // return traits::eof on failure
+            return Traits::eof();
+        }
+        // returning any value != traits::eof is fine on success, so invert eof
+        return !(Traits::eof());
+    }
+};
+
+/* @brief Gdcm log message => std::ostream => Inviwo LogCentral
+
+simple line oriented logging ostream class utilizing streambuf_lineout
+not full-featured, /see streambuf_lineout
+TODO: check standard conformance
+*/
+template <LogLevel loglevel>
+class ivw_gdcm_log_stream : public std::ostream {
+public:
+    // std::ios(0) constructor?
+    ivw_gdcm_log_stream() : std::ostream(&lnstreambuf) {}
+
+    virtual ~ivw_gdcm_log_stream() {
+        // calls sync to clean things up
+        lnstreambuf.pubsync();
+    }
+
+private:
+    streambuf_lineout<loglevel, char, std::char_traits<char>> lnstreambuf;
+};
+
+void enableGdcmLogging(LogVerbosity verbosity) {
+    static ivw_gdcm_log_stream<LogLevel::Info> debug_log_stream;
+    static ivw_gdcm_log_stream<LogLevel::Warn> warning_log_stream;
+    static ivw_gdcm_log_stream<LogLevel::Error> error_log_stream;
+
+    // reset logging/tracing in gdcm
+    gdcm::Trace::Trace();
+
+    switch (verbosity) {
+        case LogVerbosity::Info:
+            gdcm::Trace::DebugOn();
+            gdcm::Trace::SetDebugStream(debug_log_stream);
+            [[fallthrough]];
+        case LogVerbosity::Warn:
+            gdcm::Trace::WarningOn();
+            gdcm::Trace::SetWarningStream(warning_log_stream);
+            [[fallthrough]];
+        case LogVerbosity::Error:
+            gdcm::Trace::ErrorOn();
+            gdcm::Trace::SetErrorStream(error_log_stream);
+            break;
+        case LogVerbosity::None:
+        default:
+            break;
+    }
+
+    /*
+#if defined(IVW_DEBUG)
+    gdcmDebugMacro("GDCM debug output available.");
+    gdcmWarningMacro("GDCM warning output available.");
+    gdcmErrorMacro("GDCM error output available.");
+#endif
+    */
+}
+
+// =====================================================================
+// logging functionality of libtiff - using variable argument lists and snprintf
+// =====================================================================
+
+// sprintf for msvc - http://stackoverflow.com/questions/2915672/snprintf-and-visual-studio-2010
+#if defined(_MSC_VER) && _MSC_VER < 1900
```

Comment:
We require 1910 now

Reply:
yes, but that workaround is only needed up to VS 2010

Reply:
> 
> 
> We require 1910 now

Nice. Then we should update our readme etc

Reply:
Have a pr with the new requirements on the way